tests/network/vmi_infosource.go

```diff
@@ -0,0 +1,146 @@
+				Bridge: &kvirtv1.InterfaceBridge{},
+			},
+			MacAddress: "02:00:a0:a0:a0:a0",
+		}
+
+		linuxBridgeNetwork := kvirtv1.Network{
+			Name: secondaryLinuxBridgeInterface.Name,
+			NetworkSource: kvirtv1.NetworkSource{
+				Multus: &kvirtv1.MultusNetwork{
+					NetworkName: secondaryLinuxBridgeInterface.Name,
+				},
+			},
+		}
+
+		BeforeEach(func() {
+			By("Create NetworkAttachmentDefinition")
+			nad := generateNetworkAttachmentDefinition()
+			_, err := virtClient.NetworkClient().K8sCniCncfIoV1().NetworkAttachmentDefinitions(util.NamespaceTestDefault).Create(context.TODO(), nad, metav1.CreateOptions{})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			defaultBridgeInterface := libvmi.InterfaceDeviceWithBridgeBinding(defaultInterfaceName)
+			vmiSpec := libvmi.NewFedora(
+				libvmi.WithInterface(*libvmi.InterfaceWithMac(&defaultBridgeInterface, originalMac)),
+				libvmi.WithNetwork(kvirtv1.DefaultPodNetwork()),
+				libvmi.WithInterface(secondaryLinuxBridgeInterface),
+				libvmi.WithNetwork(&linuxBridgeNetwork),
+				libvmi.WithCloudInitNoCloudUserData("#!/bin/bash\nsudo ip link set dev eth0 address "+updatedMac+"\n", false))
+
+			vmi, err = virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmiSpec)
+			Expect(err).NotTo(HaveOccurred())
+			tests.WaitForSuccessfulVMIStart(vmi)
+			tests.WaitAgentConnected(virtClient, vmi)
+		})
+
+		It("should have the expected entries in vmi status when mac is changed", func() {
+			expectedInterfaces := []kvirtv1.VirtualMachineInstanceNetworkInterface{
+				{
+					InfoSource: "domain",
+					MAC:        originalMac,
+					Name:       defaultInterfaceName,
+				},
+				{
+					InfoSource:    "domain, guest-agent",
+					InterfaceName: "eth1",
+					MAC:           secondaryLinuxBridgeInterface.MacAddress,
+					Name:          secondaryLinuxBridgeInterface.Name,
+				},
+				{
+					InfoSource:    "guest-agent",
+					InterfaceName: "eth0",
+					MAC:           updatedMac,
+				},
+			}
+
+			Eventually(func() error {
+				vmiObj, err := virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+
+				if len(vmiObj.Status.Interfaces) != len(expectedInterfaces) {
+					return fmt.Errorf("interfaces doesn't match len error")
+				}
+
+				for i := range vmiObj.Status.Interfaces {
+					expectedInterfaces[i].IP = vmiObj.Status.Interfaces[i].IP
+					expectedInterfaces[i].IPs = vmiObj.Status.Interfaces[i].IPs
+				}
+
+				if reflect.DeepEqual(expectedInterfaces, vmiObj.Status.Interfaces) {
+					return nil
+				}
+				return fmt.Errorf("interfaces doesn't match\n%v\n%v", expectedInterfaces, vmiObj.Status.Interfaces)
+			}, 140*time.Second, 2*time.Second).Should(Succeed())
+		})
+	})
+})
+
+func generateNetworkAttachmentDefinition() *k8snetworkplumbingwgv1.NetworkAttachmentDefinition {
+	return &k8snetworkplumbingwgv1.NetworkAttachmentDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "secondary",
```

Comment:
This is too generic name, it may be used by other tests and then it will cause a mess.

Reply:
done